database/migrations/2023_10_05_153934_sync_permissions_and_roles.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+use App\Enums\Role as EnumsRole;
+use App\Models\Permission;
+use App\Models\Role;
+use App\Support\PermissionRepository;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\Cache;
+use Spatie\Permission\PermissionRegistrar;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Cache::forget('permissions');
+
+        $permissions = PermissionRepository::all()->map(static fn ($permission) => [
+            'name' => $permission,
+            'guard_name' => 'admin',
+            'created_at' => now(),
+            'updated_at' => now(),
+        ])->all();
+
+        // Sync permissions by name
+        Permission::upsert($permissions, ['name', 'guard_name'], []);
+
+        /**
+         * @var array<string, string> $roles
+         */
+        $roles = config('permission.roles');
+
+        $roles = collect($roles)->map(static fn ($permissions, $role) => [
+            'name' => $role,
+            'guard_name' => 'admin',
+            'created_at' => now(),
+            'updated_at' => now(),
+        ])->all();
+
+        Role::upsert($roles, ['name', 'guard_name'], []);
```

Comment:
can we assign all the permissions to the roles in the migration as well? then we don't have to go in and assign them manually when reseeding locally + will make it easier in production as well if everything is setup correctly immediately

Reply:
@ItsANameToo the problem with doing that is that is going to override any previous permissions assigned by the user (maybe thats fine)

locally the seeders still run the permissions seeder that adds the default permissions so should not be a problem locally so depends on whether we want or not to override any previously assigned permission in the migration, in my mind feels cleaner to keep that on the seeder but also not a problem since there permissions are going to be defined for the first time, what do you think?

Reply:
right now there are no customisations made, so we can safely adjust them still. I'd have the migration overwrite it for now as it won't affect us on demo/prod and we can re-assess this for subsequent releases as needed

Reply:
and it makes it less error-prone than manually assigning the permissions to the roles